### ISSUE TITLE
Crash with lambda combined with a switch expression

### ISSUE THREAD

**Issue by msridhar** [uber/NullAway#524]
Test case:
```java
  @Test
  public void testSwitchExprLambda() {
    defaultCompilationHelper
        .addSourceLines(
            "SwitchExpr.java",
            "package com.uber;",
            "import java.util.function.Function;",
            "class SwitchExpr {",
            "  int i;",
            "  public void testSwitchExprLambda() {",
            "    Function<SwitchExpr,Object> f = (s) -> switch (s.i) { case 3, 4, 5 -> new Object(); default -> \"hello\"; };",
            "  }",
            "}")
        .doTest();
  }
```

Gives us a crash:
```
  Caused by: java.lang.AssertionError: case visitor is implemented in SwitchBuilder
  	at org.checkerframework.nullaway.dataflow.cfg.builder.CFGTranslationPhaseOne.visitCase(CFGTranslationPhaseOne.java:2407)
  	at org.checkerframework.nullaway.dataflow.cfg.builder.CFGTranslationPhaseOne.visitCase(CFGTranslationPhaseOne.java:193)
  	at jdk.compiler/com.sun.tools.javac.tree.JCTree$JCCase.accept(JCTree.java:1357)
  	at jdk.compiler/com.sun.source.util.TreePathScanner.scan(TreePathScanner.java:86)
  	at org.checkerframework.nullaway.dataflow.cfg.builder.CFGTranslationPhaseOne.scan(CFGTranslationPhaseOne.java:491)
  	at org.checkerframework.nullaway.dataflow.cfg.builder.CFGTranslationPhaseOne.scan(CFGTranslationPhaseOne.java:193)
  	at jdk.compiler/com.sun.source.util.TreeScanner.scan(TreeScanner.java:111)
```

Reported in https://github.com/uber/NullAway/issues/289#issuecomment-999757311 by @lessthanoptimal.  Looks like an issue with Checker Framework; will report upstream.

**Comment by lazaroclapp** [comment]
Probably worth a known-issue comment in our changelog if we plan to release NullAway 0.9.3 with this issue outstanding. Something along the lines of:

> Experimental JDK17 support, know crash on lambdas with switch expressions as body, see `#524`

**Comment by msridhar** [comment]
Yes I think we should go ahead and release, and note the issue. Checker Framework may not have another release until end of January. Also we should note you can work around this crash temporarily with `@SuppressWarnings`

**Comment by msridhar** [comment]
The upstream Checker Framework issue typetools/checker-framework#4995 has been fixed, and I confirmed that with this fix, NullAway does not crash anymore on the given test case.

If anyone would like to test the fix locally, you can do the following:

1. Clone the [Checker Framework](https://github.com/typetools/checker-framework) and run `./gradlew publishToMavenLocal`.
2. Clone NullAway, and check out the [`lambda-switch-crash` branch](https://github.com/uber/NullAway/tree/lambda-switch-crash).  Then run `./gradlew publishToMavenLocal`.
3. Update the NullAway version in your project to `0.9.5-SNAPSHOT` (adding `mavenLocal()` as a repository, _before_ the Sonatype snapshots repo if it is there) and run the build.

@lessthanoptimal any chance you could give the above a try on your code base?  If this eliminates the crashes for you, I can ask the Checker Framework folks if they can cut a new release early, so we can then create a NullAway release that includes the fix.  And if you find another crash, we can try to get that fixed ASAP.

**Comment by lessthanoptimal** [comment]
Just tested and it works!

**Comment by msridhar** [comment]
@lessthanoptimal this fix is now released in NullAway 0.9.5

**Comment by lessthanoptimal** [comment]
thanks I'll upgrade!
